Day3/day3.py

```diff
@@ -0,0 +1,92 @@
+from re import match
+
+data = open('Day3/input.txt', 'r').read().split('\n')
+testdata = open('Day3/testinput.txt', 'r').read().split('\n')
+
+
+class DataArray:
+    def __init__(self):
+        self.data = []
+        self.arrayOfNumbers = []
+        self.arrayOfSpecial = []
+        self.finalResult = 0
+
+    def __getArray__(self, indexx, indexy): 
+        return self.data[indexx][indexy]
+
+    def getData(self, data): # init array :D
+        for line in data:
+            self.data.append(list(line))
+        self.maxrow = len(self.data)
+        self.maxcollum = len(self.data[0])
+
+    def getArrayOfNumbers(self): 
+        for row in range(self.maxrow):
+            for col in range(self.maxcollum):
+                if (self.data[row][col].isnumeric()):
+                    self.arrayOfNumbers.append([row, col])        
+    
+    def succesNeighbor(self, iterator):
+        try:
+            if (self.arrayOfNumbers[iterator + 1][1] == (self.arrayOfNumbers[iterator][1] + 1)):
+                    self.succesNeighbor(iterator + 1)
+            else:    
+                self.resolvingNumber(iterator, True)
+        except IndexError:
+            if(len(self.arrayOfNumbers) > 0):
```

Comment:
you could wrap all in IndexError in a method and use it in notSuccessNeighbour as well